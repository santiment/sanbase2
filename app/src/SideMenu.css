.menu-list ul {
  margin-top: 24px;
  padding: 0;
  list-style: none;
}

.menu-list li {
  margin: 0;
  padding: 10px 22px;
  display: flex;
  align-items: center;
  cursor: pointer;
}

.menu-list li:hover {
  background-color: #4f5b69;
  transition: all 1s ease;
}

.menu-list li a {
  text-decoration: none;
  color: rgba(255,255,255,0.5);
  display: list-item;
  list-style-type: none;
}

.menu-list .fa {
  margin-right: 12px;
}

.menu-list .sub-menu {
  margin: 0;
}

.menu-list .sub-menu li {
  margin: 0;
  padding-left: 48px;
}

.menu-list .sub-menu li:hover {
  background-color: rgba(255,255,255,0.2);
}

.menu-list .sub-menu li .fa {
  font-size: 6px;
  margin-right: 10px;
}

.user-auth-control {
  padding-left: 20px;
}

.user-auth-control a {
  cursor: pointer;
}

.user-auth-control button {
  width: 90%;
}
