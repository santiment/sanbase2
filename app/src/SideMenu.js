import React from 'react'
import {
  compose,
  pure
} from 'recompose'
import { connect } from 'react-redux'
import { NavLink as Link } from 'react-router-dom'
import logo from './assets/logo_sanbase.png'

export const SideMenu = ({
  user,
  loading,
  logout
}) => (
  <div className='side-menu'>
    <div className='brand'>
      <img
        src={logo}
        width='115'
        height='22'
        alt='SANbase' />
    </div>
    <i
      className='fa fa-bars fa-2x toggle-btn'
      data-toggle='collapse'
      data-target='#menu-content' />
    <div className='menu-list'>
      <ul id='menu-content' className='menu-content collapse out'>
        <li>
          Dashboard (tbd)
        </li>
        <li data-toggle='collapse' data-target='#products'>
          Data-feeds
        </li>
        <ul className='sub-menu' id='products'>
          <li>Overview (tbd)</li>
          <li>
            <Link
              activeClassName='selected'
              to='/cashflow'>
              Cash Flow
            </Link>
          </li>
        </ul>
        <li>
          Signals
        </li>
        <li>
          Roadmap
        </li>
      </ul>
    </div>

    <br />

    {user.username
      ? <div className='user-auth-control'>
        You are logged in!
        <br />
        <button
          onClick={logout}>
          Log out
        </button>
      </div>
      : <div className='user-auth-control'>
        <Link to={'/login'}>
          Log in
        </Link>
      </div>
      }
  </div>
)

const mapStateToProps = state => {
  return {
    user: state.user.data,
    loading: state.user.isLoading
  }
}

const mapDispatchToProps = dispatch => {
  return {
    logout: () => {
      dispatch({
        type: 'SUCCESS_LOGOUT'
      })
    }
  }
}

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  pure
)

export default enhance(SideMenu)
