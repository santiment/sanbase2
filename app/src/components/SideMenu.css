.user-auth-control {
  padding-left: 20px;
}

.user-auth-control a {
  cursor: pointer;
}

.user-auth-control button {
  width: 90%;
}

.night-mode .user-auth-control .button {
  background-color: #1e2024;
}

.night-mode .user-auth-control .button i {
  color: #a0a0a0;
}
