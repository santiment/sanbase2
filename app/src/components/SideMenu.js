import React from 'react'
import { withRouter } from 'react-router-dom'
import {
  compose,
  pure
} from 'recompose'
import { connect } from 'react-redux'
import 'font-awesome/css/font-awesome.css'
import logo from '../assets/logo_sanbase.png'
import AppMenu from './AppMenu'
import AuthControl from './AuthControl'
import './SideMenu.css'

export const SideMenu = ({
  user,
  loading,
  logout,
  history
}) => (
  <div className='side-menu'>
    <div
      onClick={() => history.push('/')}
      className='brand'>
      <img
        src={logo}
        width='115'
        height='22'
        alt='SANbase' />
    </div>
    <AppMenu
      handleNavigation={nextRoute => {
        history.push(`/${nextRoute}`)
      }} />
    <AuthControl
      login={() => history.push('/login')}
      user={user}
      logout={logout} />
  </div>
)

const mapStateToProps = state => {
  return {
    user: state.user.data,
    loading: state.user.isLoading
  }
}

const mapDispatchToProps = dispatch => {
  return {
    logout: () => {
      dispatch({
        type: 'SUCCESS_LOGOUT'
      })
    }
  }
}

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withRouter,
  pure
)

export default enhance(SideMenu)
