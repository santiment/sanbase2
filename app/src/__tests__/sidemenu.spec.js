/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { SideMenu } from './../SideMenu'

describe('SideMenu container', () => {
  it('it should render correctly', () => {
    const user = {
      username: null,
      token: null
    }
    const loading = false
    const login = shallow(<SideMenu user={user} loading={loading} />)
    expect(toJson(login)).toMatchSnapshot()
  })

  it('it should show logout button', () => {
    const user = {
      username: 'user',
      token: 'asdfasdfi'
    }
    const loading = false
    const login = shallow(<SideMenu user={user} loading={loading} />)
    expect(toJson(login)).toMatchSnapshot()
    expect(login.find('.user-auth-control button').text()).toEqual('Log out')
  })

  it('it should show login button', () => {
    const user = {
      username: '',
      token: ''
    }
    const loading = false
    const login = shallow(<SideMenu user={user} loading={loading} />)
    expect(toJson(login)).toMatchSnapshot()
    expect(login.find('.user-auth-control NavLink').contains('Log in')).toEqual(true)
  })
})
