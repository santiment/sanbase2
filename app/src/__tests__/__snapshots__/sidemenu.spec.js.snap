// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu container it should render correctly 1`] = `
<div
  className="side-menu"
>
  <div
    className="brand"
  >
    <img
      alt="SANbase"
      height="22"
      src="logo_sanbase.png"
      width="115"
    />
  </div>
  <AppMenu
    handleNavigation={[Function]}
  />
  <AuthControl
    login={[Function]}
    user={
      Object {
        "token": null,
        "username": null,
      }
    }
  />
</div>
`;

exports[`SideMenu container it should show login button 1`] = `
<div
  className="side-menu"
>
  <div
    className="brand"
  >
    <img
      alt="SANbase"
      height="22"
      src="logo_sanbase.png"
      width="115"
    />
  </div>
  <AppMenu
    handleNavigation={[Function]}
  />
  <AuthControl
    login={[Function]}
    user={
      Object {
        "token": "",
        "username": "",
      }
    }
  />
</div>
`;

exports[`SideMenu container it should show logout button 1`] = `
<div
  className="side-menu"
>
  <div
    className="brand"
  >
    <img
      alt="SANbase"
      height="22"
      src="logo_sanbase.png"
      width="115"
    />
  </div>
  <AppMenu
    handleNavigation={[Function]}
  />
  <AuthControl
    login={[Function]}
    user={
      Object {
        "token": "asdfasdfi",
        "username": "user",
      }
    }
  />
</div>
`;
